### PR TITLE
Ebonilla/sy 749 opc ua read task cleanup and tests

### PR DESCRIPTION
**Pull request by emilbon99**

### CODE REVIEW

**Review by LeonLiur — APPROVED** [review]
Good after some minor changes

File: cesium/internal/virtual/write_test.go [hunk 1]
```diff
@@ -20,6 +21,7 @@ var _ = Describe("Write", func() {
 				DataType: telem.TimeStampT,
 				Virtual:  true,
 			},
+			FS: xfs.NewMem(),
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply cesium/internal/virtual/write_test.go:24]: FS implementation is still checked for in the config but does not apply to any test cases. Feel free to adjust as you build out future test suites

File: client/cpp/framer/framer.h [hunk 2]
```diff
@@ -170,9 +170,9 @@ class Streamer {
 };
 
 enum WriterMode : uint8_t {
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply client/cpp/framer/framer.h:172]: Yes, you can set the mode when opening the writer

File: client/cpp/hardware/hardware.cpp [hunk 3]
```diff
@@ -159,6 +179,62 @@ std::pair<Task, freighter::Error> TaskClient::retrieve(std::uint64_t key) const
...
+    req.set_rack(rack);
+    req.mutable_names()->Add(names.begin(), names.end());
+    auto [res, err] = task_retrieve_client->send(RETRIEVE_MODULE_ENDPOINT, req);
+    if (err) return {std::vector<Task>(), err};
+    std::vector<Task> tasks = {res.tasks().begin(), res.tasks().end()};
+    return {tasks, err};
+}
+
+
+std::pair<Task, freighter::Error> TaskClient::retrieveByType(
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply client/cpp/hardware/hardware.cpp:209]: We need to standardize the convention we use. I'll leave this to @Lham42 as a chore

File: client/py/examples/control/tpc/change_detection.py [hunk 4]
```diff
@@ -10,7 +10,9 @@
         elif value["fuel_pt_1"] < 20 and above_threshold:
             client.ranges.create(
                 name=f"Fuel Above Threshold - " + str(above_threshold)[11:19],
-                time_range=sy.TimeRange(start=above_threshold, end=value["daq_time"][-1]),
-                color="#BADA55"
+                time_range=sy.TimeRange(
+                    start=above_threshold, end=value["daq_time"][-1]
+                ),
+                color="#BADA55",
```
[the remark this answers is not in the document]
Comment (emilbon99) [reply client/py/examples/control/tpc/change_detection.py:16]: Ha

**Review by Lham42 — APPROVED** [review]

**Pull request merged**